tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -939,6 +939,61 @@ def binned_entropy(x, max_bins):
 
 @set_property("fctype", "aggregate_with_parameters")
 @not_apply_to_raw_numbers
+def sample_entropy(x, tolerance=None):
+    """
+    Calculate and return sample entropy of x.
+
+    References:
+    ----------
+    [1] http://en.wikipedia.org/wiki/Sample_Entropy
+    [2] http://physionet.incor.usp.br/physiotools/sampen/
+    [3] Madalena Costa, Ary Goldberger, CK Peng. Multiscale entropy analysis of biological signals
+    
+    :param x: the time series to calculate the feature of
+    :type x: pandas.Series
+    :param tolerance: normalization factor; equivalent to the common practice of expressing the tolerance as r times the standard deviation
+    :type tolerance: float
+    :return: the value of this feature
+    :return type: float
+    """
+    sample_length = 1 # number of sequential points of the time series
+
+    if tolerance is None:
+        tolerance = 0.1 * np.std(time_series)
+
+    n = len(time_series)
+    prev = np.zeros((1, n))
+    curr = np.zeros((1, n))
+    A = np.zeros((sample_length, 1))  # number of matches for m = [1,...,template_length - 1]
+    B = np.zeros((sample_length, 1))  # number of matches for m = [1,...,template_length]
+
+    for i in xrange(n - 1):
+        nj = n - i - 1
+        ts1 = time_series[i]
+        for jj in xrange(nj):
+            j = jj + i + 1
+            if abs(time_series[j] - ts1) < tolerance:  # distance between two vectors
+                curr[0, jj] = prev[0, jj] + 1
```

Comment:
Curr and previous stay 1 dimensional the whole time, why make them two dimensional?